performance_manager/tests/test_postgres.py

```diff
@@ -30,21 +76,332 @@ def set_environment_variables() -> Iterator[None]:
+        position_size = positions.shape[0]
+        assert positions.shape[1] == 9
+
+        # check that the types can be set correctly
+        positions = transform_vp_datatypes(positions)
+        assert positions.shape[1] == 9
+        assert position_size == positions.shape[0]
+
+        # check that it can be combined with the static schedule
+        positions = join_vp_with_gtfs_static(positions, db_manager)
+        assert positions.shape[1] == 11
+        assert position_size == positions.shape[0]
+
+        positions = transform_vp_timestamps(positions)
+        assert positions.shape[1] == 12
+        assert positions.shape[0] < position_size
+
+        trip_updates = get_and_unwrap_tu_dataframe(files["tu_paths"])
+        trip_update_size = trip_updates.shape[0]
+        assert trip_updates.shape[1] == 9
+
+        trip_updates = join_tu_with_gtfs_static(trip_updates, db_manager)
+        assert trip_updates.shape[1] == 11
+        assert trip_update_size == trip_updates.shape[0]
+
+        trip_updates = reduce_trip_updates(trip_updates)
+        assert trip_update_size > trip_updates.shape[0]
+
+        events = combine_events(positions, trip_updates)
+
+        ve_columns = [c.key for c in VehicleEvents.__table__.columns]
+        expected_columns = set(ve_columns) - {"pk_id", "updated_on"}
+        assert len(expected_columns) == len(events.columns)
+
+        missing_columns = set(events.columns) - expected_columns
+        assert len(missing_columns) == 0
+
+        upload_to_database(events, db_manager)
+
+    check_logs(caplog)
+
+
+def test_vp_only(
+    db_manager: DatabaseManager, caplog: pytest.LogCaptureFixture
+) -> None:
+    """
+    test the vehicle positions can be updated without trip updates
+    """
+    caplog.set_level(logging.INFO)
+
+    db_manager.truncate_table(VehicleEvents)
+    db_manager.execute(
+        sa.delete(MetadataLog.__table__).where(
+            ~MetadataLog.path.contains("FEED_INFO")
+        )
+    )
+
+    paths = [
+        p
+        for p in test_files()
+        if "RT_VEHICLE_POSITIONS" in p and ("hourt=10" in p or "hour=11" in p)
+    ]
+    db_manager.add_metadata_paths(paths)
+
+    process_gtfs_rt_files(db_manager)
+
+    check_logs(caplog)
+
+
+def test_tu_only(
+    db_manager: DatabaseManager, caplog: pytest.LogCaptureFixture
+) -> None:
+    """
+    test the trip updates can be processed without vehicle positions
+    """
+    caplog.set_level(logging.INFO)
+
+    db_manager.truncate_table(VehicleEvents)
+    db_manager.execute(
+        sa.delete(MetadataLog.__table__).where(
+            ~MetadataLog.path.contains("FEED_INFO")
+        )
+    )
+
+    paths = [
+        p
+        for p in test_files()
+        if "RT_TRIP_UPDATES" in p and ("hourt=10" in p or "hour=11" in p)
+    ]
+
+    db_manager.add_metadata_paths(paths)
+
+    process_gtfs_rt_files(db_manager)
+
+    check_logs(caplog)
+
+
+def test_vp_and_tu(
```

Comment:
test that the processing works in general.

i'd really like to add a check that pulls data from the db and compares it to values that phil gave me for the date all the parquet files are from.